aws/resource_aws_batch_job_definition.go

```diff
@@ -0,0 +1,212 @@
+package aws
+
+import (
+	"fmt"
+
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/batch"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsBatchJobDefinition() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBatchJobDefinitionCreate,
+		Read:   resourceAwsBatchJobDefinitionRead,
+		Delete: resourceAwsBatchJobDefinitionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateBatchName,
+			},
+			"container_properties": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
+				},
+				DiffSuppressFunc: suppressEquivalentJsonDiffs,
+				ValidateFunc:     validateAwsBatchJobContainerProperties,
+			},
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+				Elem:     schema.TypeString,
+			},
+			"retry_strategy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"attempts": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.JobDefinitionTypeContainer}, true),
+			},
+			"revision": {
+				Type:     schema.TypeInt,
+				Optional: true,
```

Comment:
The `revision` here is still `Optional`, but I will take care of that prior to merging.